<?xml version="1.0"?>
<launch>
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name ="launch_prefix" value="" />
  <arg if="$(arg debug)" name ="launch_prefix" value="gdb --ex run --args" />
  
  <arg name="debug-remote" default="false"/>
  <arg unless="$(arg debug-remote)" name ="waittime" value="0" />
  <arg if="$(arg debug-remote)" name ="waittime" value="30" />
  
  <arg name="youBotHasBase" default="true" />
  <arg name="youBotHasArms" value="true" /> 
  <arg name="youBotDriverCycleFrequencyInHz" value="50.0" />
  <arg name="trajectoryActionServerEnable" value="true" />
  <arg name="trajectoryPositionGain" value="5.0" />
  <arg name="trajectoryVelocityGain" value="0.0" />
  <arg name="dualArm" default="false" />
  <arg name="youBotBaseName" value="youbot-base"/>
  <arg name="youBotArmName1" value="youbot-manipulator"/>
  <arg name="youBotArmName2" value="youbot-manipulator-dual-arm"/>

  <!-- Set Params -->
  <param name="youBotHasBase" type="bool" value="$(arg youBotHasBase)" />
  <param name="youBotHasArms" type="bool" value="$(arg youBotHasArms)" /> 
  <param name="youBotDriverCycleFrequencyInHz" type="double" value="$(arg youBotDriverCycleFrequencyInHz)" />
  <param name="trajectoryActionServerEnable" type="bool" value="$(arg trajectoryActionServerEnable)" />
  <param name="trajectoryPositionGain" type="double" value="$(arg trajectoryPositionGain)" />
  <param name="trajectoryVelocityGain" type="double" value="$(arg trajectoryVelocityGain)" />
  <param name="youBotBaseName" type="str" value="$(arg youBotBaseName)" />
  <param name="youBotArmName1" type="str" value="$(arg youBotArmName1)" />

  <group if="$(arg dualArm)">
    <param name="youBotArmName2" type="str" value="$(arg youBotArmName2)" />
    <node name="youbot_wrapper" launch-prefix="$(arg launch_prefix)" pkg="youbot_common" type="youbot_wrapper" args="$(arg waittime)" output="screen" />
  </group>
  <group unless="$(arg dualArm)">
    <node name="youbot_wrapper" pkg="youbot_common" launch-prefix="$(arg launch_prefix)" type="youbot_wrapper" args="$(arg waittime)" output="screen">
<!--      <remap from="base/joint_states" to="joint_states" />  --> 
<!--      <remap from="arm_1/joint_states" to="joint_states" /> -->
    </node>
  </group>
</launch>
